week-2-wide-perspective/w02e04-dungeons-and-the-princess/src/rescue-mission.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { describe, it, expect, beforeAll, beforeEach, vi } from "vitest";
+import { Castle } from './model/castle'
+import { Dungeon } from './model/dungeon'
+import { Person } from './model/person'
+import { Underground } from './model/underground'
+import { createRescueMission } from "./helper/mission";
+
+describe('Recuse mission', () => {
+    let princess: Person;
+    let hades: Underground;
+    let cave: Dungeon;
+    let kingdom: Castle;
+
+    let knightJohn: Person
+    let knightBrienne: Person
+    let kingFrancis: Person
+    let peasantMathew: Person
+    let queenBianca: Person
+    let knightBruce: Person
+
+    beforeEach(() => {
+        princess = new Person('Fiona', 'princess')
+        hades = new Underground(princess)
+        cave = new Dungeon(hades)
+        kingdom = new Castle(cave)
+
+        knightJohn = new Person('John', 'knight')
+        knightBrienne = new Person('Brienne', 'knight')
+        kingFrancis = new Person('Francis', 'king')
+        peasantMathew = new Person('Mathew', 'peasant')
+        queenBianca = new Person('Bianca', 'queen')
+        knightBruce = new Person('Bruce', 'knight')
+    })
+
+    it("three knights should rescue the prisoner", () => {
+        const spy = vi.spyOn(princess, 'sayThanks')
+        createRescueMission([knightBruce, knightBrienne, knightJohn], hades, kingdom)
+        expect(spy).toHaveBeenCalledTimes(1);
+    })
```

Comment:
To jest jedyny wyżej poziomowy test funkcjonalny.

Powinieneś dorzucić jeszcze pozostałe `Rescue Mission` i upewnić się że rzucają.

Poniżej masz `Unit testy` danych funkcjonalności.
Jednak fajnie by było potwierdzić to założenie z góry.

Na górze w 
> "three knights should rescue the prisoner"

Sprawdzasz wesołą ścieżkę - zobacz też ze 2-3 smutne.